sysbuild/Kconfig.zip

```diff
@@ -44,6 +44,46 @@ config DFU_ZIP_BLUETOOTH_MESH_METADATA
 	help
 	  Includes metadata for bluetooth mesh.
 
+if DFU_ZIP_BLUETOOTH_MESH_METADATA
+
+choice DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID
+	prompt "Source of Firmware ID to include in metadata file"
+	default DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM
+
+config DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM
+	bool "Use a custom Firmware ID"
+	help
+	  The value to include as Firmware ID in the Bluetooth Mesh metadata
+	  file is set using DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM_HEX.
+	  Note: You are required to make sure that the Firmware ID used when
+	  initializing the DFU Server matches the ID in the metadata file.
+
+config DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_MCUBOOT_VERSION
+	bool "Use a Firmware ID generated from the MCUboot version"
+	help
+	  Includes a Firmware ID in the Bluetooth Mesh metadata file consisting
+	  of BT_COMPANY_ID and the MCUboot version.
+	  Note: You are required to make sure that the Firmware ID used when
+	  initializing the DFU Server matches the ID in the metadata file.
+
+endchoice
+
+if DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM
+
+config DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM_HEX
+	string "HEX string to use as Firmware ID"
+	help
+	  The raw value of this option is included as the Firmware ID in the
+	  Bluetooth Mesh metadata file. This must be a valid hex string, and
+	  must at minimum begin with a Company ID. The rest of the string is
```

Comment:
We should mention endianness here.

Reply:
For the Company ID, then? The spec does not really care about the contents (and by extension, endianness) of the vendor-specific data. (I mean, it may perhaps _claim_ that it should be little endian, depending on your reading of it, but there is no way for anyone to verify this without knowing the scheme)

Reply:
I guess for the entire value? The rules of the access layer apply here:
```
3.7.1 Endianness
All multiple-octet numeric values in this layer shall be little-endian as described in Section 3.1.1.2.
```

> there is no way for anyone to verify this without knowing the scheme)

Unless (I guess) there is some 3rd-party UI (e.g. 3rd party mobile) that shows to a user (for some reasons) the version information as a string.

Reply:
> [...] shows to a user (for some reasons) the version information as a string.

As a _hex_ string? Because in that case it would still not be possible to tell the endianness, it's still just a row of bytes?

Reply:
3.7.1 was my entire point about "it may perhaps _claim_ that it should be little endian". I mean, the spec says so, but no-one will ever be able to check this except for the vendor who designs the layout of the vendor-specific information. This is, since it is vendor-specific, not an interoperable piece of data, so IMO the spec doesn't really care what the bytes are even though it might claim to do so.